setup.py

```diff
@@ -69,6 +69,42 @@
     ],
     package_data = {'nltk': ['test/*.doctest', 'VERSION']},
     install_requires = ['six>=1.10.0'],
+    extras_require = {
```

Comment:
that's great! I have two comments though.
 
1. Unless it's really known that nltk doesn't work with a lower version of a package, I would not set the minimum version.  For example, NLTK doesn't make use of numpy's functionality that is available only from version 1.8.0, I guess.

2. The list should be easy maintainable. Right now, gensim is listed under ``all`` and ``machine_learning``. Maybe, it worth constructing the dictionary of extras outside of the setup() call. Then ``all`` is the union of all extras, something like:
```python
extras_require = {
    'machine_learning': [...],
    'plot ': [...],
    ...
}

extras_require['all'] = set(itertools.chain.from_iterable(extras_require.values()))

```